### DISCUSSION TITLE
Month frequency data leads to invalid frequency

### DISCUSSION THREAD

**Discussion by yarnabrina** [alan-turing-institute/sktime#4666] [Q&A]
Consider this code:

```python
import numpy
import pandas
from sktime.forecasting.naive import NaiveForecaster

numpy.random.seed(0)

times = pandas.to_datetime(["2024-01-01", "2024-02-01", "2024-03-01", "2024-04-01"])
targets = numpy.random.normal(50, 5, 4)

dataset = pandas.Series(targets, index=times)

NaiveForecaster().fit(dataset).predict([1, 2])
```

This leads to the following error:

> ValueError: Invalid frequency. Please select a frequency that can be converted to a regular `pd.PeriodIndex`. For other frequencies, basic arithmetic operation to compute durations currently do not work reliably.

This works if I change `times` to day or week frequency:

```python
times = pandas.to_datetime(["2024-01-01", "2024-01-02", "2024-01-03", "2024-01-04"])
```

or

```python
times = pandas.to_datetime(["2024-01-01", "2024-01-08", "2024-01-15", "2024-01-22"])
```

This seems related to https://github.com/sktime/sktime/pull/582. However, it is possible to actually possible to convert this (month) case to `PeriodIndex`:

```pycon
>>> 
>>> times = pandas.to_datetime(["2024-01-01", "2024-02-01", "2024-03-01", "2024-04-01"])
>>> times.to_period()
PeriodIndex(['2024-01', '2024-02', '2024-03', '2024-04'], dtype='period[M]')
>>> 
```

Question: as a user, how should I modify the inputs to make it work? Since the exact same format works for other frequencies, a consistent format for all frequency/mtype will be very helpful.

**Comment by yarnabrina** [comment]
@fkiraly created this as discussion as I am not sure if you'll consider this as a `pandas` bug as the linked PR indicates, or a potential enhancement request in `fh`.

**Reply by fkiraly** [reply]
hm, could be considered both a bug or af eature request, as there is not a stated contract for `fh` in case of irregular time index, and I'm not sure what the expected contract is. It's probably a reasonable expectation that the above code doesn't break...

What would you advise?

**Reply by yarnabrina** [reply]
I was thinking more towards feature request as it'd support one more type of frequency. But I just discovered that it may also be a potential bug in future because `freq` attribute of `pandas.Timestamp` is deprecated, and `_coerce_to_period` is relying on it.

```python
>>> pandas.to_datetime("2024-01-01").freq
<stdin>:1: FutureWarning: Timestamp.freq is deprecated and will be removed in a future version.
```
